src/rtk/features/singleRights/singleRightsSlice.ts

```diff
@@ -185,47 +194,92 @@ const singleRightSlice = createSlice({
   },
   extraReducers: (builder) => {
     builder
+      .addCase(delegationAccessCheck.pending, (state, action) => {
+        const serviceID = action.meta.arg.serviceResource.identifier;
+        let nextServicesWithStatus;
+
+        if (
+          state.servicesWithStatus.some(
+            (s: ServiceWithStatus) => s.service?.identifier === serviceID,
+          )
+        ) {
+          nextServicesWithStatus = state.servicesWithStatus.map((sws: ServiceWithStatus) => {
+            if (sws.service?.identifier === serviceID) {
+              sws.isLoading = true;
+            }
+            return sws;
+          });
+        } else {
+          const newServiceWithStatus: ServiceWithStatus = {
+            service: action.meta.arg.serviceResource,
+            isLoading: true,
+          };
+          nextServicesWithStatus = [...state.servicesWithStatus, newServiceWithStatus];
+        }
+
+        state.servicesWithStatus = nextServicesWithStatus;
+      })
+
       .addCase(delegationAccessCheck.fulfilled, (state, action) => {
-        const serviceWithStatus: ServiceWithStatus = {
-          rightDelegationResults: action.payload,
-          service: action.meta.arg.serviceResource,
-          status: 'Delegable',
-          errorCode: '',
-        };
+        const serviceID = action.meta.arg.serviceResource.identifier;
```

Comment:
Funker koden også for når backend gir http-error eller må man legge til noe kode i BFF-en for å få det til å funke?